### PR TITLE
qualcommax: ipq60xx: add Qihoo 360V6 support

### PR DESCRIPTION
**Pull request by aiamadeus**
Specifications:
  SoC:     Qualcomm IPQ6000 1.5GHz
  RAM:     NTCC256M16ER-EK 512MiB
  Flash:   W29N01HZSINA 128MiB
  ETH:     QCA8075 (3x LAN, 1x WAN)
  WLAN1:   2.4GHz 802.11b/g/n/ax 2x2
  WLAN2:   5GHz 802.11a/n/ac/ax 2x2
  Power:   DC 12V 1.5A
  Button:  Reset, Wps
  USB:     1x 2.0

Flash instructions:
  1. Download the initramfs image, rename it to
     initramfs.itb, host it with the tftp server.
  2. Interrupt U-Boot and run these commands:
     tftpboot initramfs.itb
     bootm
  3. After openwrt boots up, use scp or luci web
     to upload sysupgrade.bin to upgrade.

### CODE REVIEW

**Comment by breeze303** [comment]
Why did it take so long to review ipq60xx official like constipation？

**Comment by aiamadeus** [comment]
@robimarko Could you take a look at this?

File: target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network [hunk 1]
```diff
@@ -29,9 +32,30 @@ ipq60xx_setup_interfaces()
...
+	local lan_mac=""
+	local wan_mac=""
+	local label_mac=""
+
+	case "$board" in
+	qihoo,360v6)
+		lan_mac=$(mtd_get_mac_ascii factory lanMac)
+		wan_mac=$(macaddr_add "$lan_mac" 1)
+		label_mac=$lan_mac
+		;;
```
Comment (robimarko) [inline target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:47]: This should be replaced via NVMEM which now can read ASCII MAC-s as well

Comment (robimarko) [reply target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:47]: It doesnt have to be, you can use NVMEM on SMEM partition tables.

Comment (aiamadeus) [reply target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:47]: Do you mean something like this?
```
	nand@0 {
		reg = <0>;
		nand-ecc-strength = <4>;
		nand-ecc-step-size = <512>;
		nand-bus-width = <8>;

		partitions {
			compatible = "qcom,smem-part";

			partition-0-factory {
				compatible = "fixed-partitions";
				label = "factory";
				read-only;
				#address-cells = <1>;
				#size-cells = <1>;

				partition@0 {
					reg = <0x0 0x40000>;

					macaddr_factory_ethaddr: lanMac {};
				};
			};
		};
	};
```

Comment (robimarko) [reply target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:47]: Close, but you cannot use fixed partitions on it:
https://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git/tree/Documentation/devicetree/bindings/mtd/partitions/qcom,smem-part.yaml?h=next-20241128#n45

Comment (aiamadeus) [reply target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:47]: Unfortunately NVMEM on SMEM partition table does not work:
```
	nand@0 {
		reg = <0>;
		nand-ecc-strength = <4>;
		nand-ecc-step-size = <512>;
		nand-bus-width = <8>;

		partitions {
			compatible = "qcom,smem-part";
			#address-cells = <1>;
			#size-cells = <1>;

			partition-factory {
				compatible = "nvmem-cells";
				#address-cells = <1>;
				#size-cells = <1>;
				label = "factory";
				read-only;

				macaddr_lan: macaddr@47 {
					compatible = "mac-base";
					reg = <0x47 0x11>;
					#nvmem-cell-cells = <1>;
				};
			};
		};
	};
```

Comment (robimarko) [reply target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:47]: Ahh, its u-boot-env variable, yeah in that case only ethaddr, eth1addr and so on are parseable for MAC-s

Comment (aiamadeus) [reply target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:47]: https://github.com/openwrt/openwrt/blob/39b0f081ecaf4fb9acc622d694dc52293a4dda1f/target/linux/generic/backport-6.6/823-v6.12-0003-nvmem-layouts-add-U-Boot-env-layout.patch#L170-L174

It's actually just "ethaddr", so can we keep using the script?

Comment (robimarko) [reply target/linux/qualcommax/ipq60xx/base-files/etc/board.d/02_network:47]: Yes, untill U-Boot driver gets converted to layouts

**Comment by robimarko** [comment]
Thanks! Rebased on top of main and merged!

**Pull request merged**